lib/ansible/modules/packaging/os/flatpak.py

```diff
@@ -0,0 +1,274 @@
+# Copyright (c) 2017 Ansible Project
+# GNU General Public License v3.0+ (see COPYING or
+# https://www.gnu.org/licenses/gpl-3.0.txt)
+
+ANSIBLE_METADATA = {
+    'metadata_version': '1.0',
+    'status': ['preview'],
+    'supported_by': 'community'}
+
+DOCUMENTATION = '''
+---
+module: flatpak
+version_added: "2.4"
+requirements:
+  - flatpak
+author:
+  - John Kwiatkoski (@jaykayy)
+short_description: Install and remove flatpaks.
+description:
+  - The flatpak module allows users to manage installation and removal of flatpaks.
+options:
+  name:
+    description:
+      - When I(state) is set to C(present), I(name) is best used as http(s) url format.
+        When set to C(absent) the same http(s) will try to remove it using the
+        name of the flatpakref. However, there is no naming standard between
+        names of flatpakrefs and what the reverse DNS name of the installed flatpak
+        will be. Given that, it is best to use the http(s) url for I(state=present)
+        and the reverse DNS I(state=absent). Alternatively reverse dns format can optimally
+        be used with I(state=absent), ex. I(name=org.gnome.gedit).
+    required: false
+    default: ''
+  remote:
+    description:
+      - The flatpak I(remote) repo to be used in the flatpak operation.
+    required: false
+    default: ''
+  state:
+    description:
+      - Set to C(present) will install the flatpak and/or I(remote).
+        Set to C(absent) will remove the flatpak and/or I(remote).
+    required: false
+    default: present
+'''
+
+EXAMPLES = '''
+ - name: Install the spotify flatpak
+   flatpak:
+    name:  https://s3.amazonaws.com/alexlarsson/spotify-repo/spotify.flatpakref
+    state: present
+
+ - name: Add the gnome remote andd install gedit flatpak
+   flatpak:
+    name: https://git.gnome.org/browse/gnome-apps-nightly/plain/gedit.flatpakref
+    remote: https://sdk.gnome.org/gnome-apps.flatpakrepo
+    state: absent
+
+ - name: Remove the gedit flatpak and remote
+   flatpak:
+    name: org.gnome.gedit
+    remote: https://sdk.gnome.org/gnome-apps.flatpakrepo
+    state: absent
+
+ - name: Remove the gedit package
+   flatpak:
+    name: org.gnome.gedit
+    state: absent
+'''
+
+RETURN = '''
+reason:
+  description: On failure, the output for the failure
+  returned: failed
+  type: string
+  sample: error while installing...
+name:
+  description: Name of flatpak given for the operation
+  returned: always
+  type: string
+  sample: https://git.gnome.org/.../gnome-apps/gedit.flatpakref
+remote:
+  description: Remote of flatpak given for the operation
+  returned: always
+  type: string
+  sample: https://sdk.gnome.org/gnome-apps.flatpakrepo
+'''
+
+
+import subprocess
+from ansible.module_utils.basic import AnsibleModule
+
+
+def install_flat(binary, flat):
+    command = "{} install -y --from {}".format(binary, flat)
+    output = flatpak_command(command)
+
+    if 'error' in output and 'already installed' not in output:
+        return 1, output
```

Comment:
I kind of don't like returning from a function from a condition. What about to define `code` variable which you assign in the condition and then return at the end of the function?

```
    ...
    output = flatpak_command(command)
    code = 0

    if 'error' in output and 'already installed' not in output:
        code = 1

    return code, output
```

The same anywhere bellow.